cmake/external_libs/eigen3.cmake

```diff
@@ -1,24 +1,16 @@
 if(UNIX)
     BoB_add_pkg_config_libraries(eigen3)
-else()
-    if(NOT TARGET Eigen3::Eigen)
-        message(FATAL_ERROR "Eigen 3 not found")
-    endif()
+elseif(TARGET Eigen3::Eigen)
     BoB_add_link_libraries(Eigen3::Eigen)
+else()
```

Comment:
Yep. I think this might only be an issue on Ubuntu 16.04.

This is one of the annoying things about CMake: it's supposed to be cross-platform but if you've got an old version of a library they might not have a bundled cmake file with it so you end up just doing a bunch of if-statements anyway (``/rant``)

Reply:
You do definitely begin to see why people get irritated with build systems and start on a fools errand to make their own....